### PR TITLE
added instruction for github opensource contributions

### PR DESCRIPTION
**Pull request by take2rohit**
This is oversimplified instructions

### CODE REVIEW

File: Intro to open source with GitHub/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# How to work on opensource contribution with Git
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:1]: Just make this 'Open Source Contributions with Git'

Comment (take2rohit) [reply Intro to open source with GitHub/README.md:1]: ok

File: Intro to open source with GitHub/README.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+## Making a contribution
+1. Fork the repo which you want to work on (eg. https://github.com/take2rohit/sih_number_plate is original repo where you want to contribute)
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:4]: '..... eg. [this repo](link) is the original repository you want to contribute to' (link in that format might be better)

File: Intro to open source with GitHub/README.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# How to work on opensource contribution with Git
+
+## Making a contribution
+1. Fork the repo which you want to work on (eg. https://github.com/take2rohit/sih_number_plate is original repo where you want to contribute) 
+1. Clone the forked repository (not the above repo link)
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:5]: numbering is 1, should be 2 (but it still renders correctly, but just change it 😂 )

Comment (akshaykulkarni07) [reply Intro to open source with GitHub/README.md:5]: Add some clarity by saying 'Clone the forked repository from your own profile (NOT the original repository)'

File: Intro to open source with GitHub/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+`git clone https://github.com/your_userid/sih_number_plate`
+3. Do your changes whatever you wish and commit
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:7]: Do the changes and commit. Let's try to be grammatically correct from the beginning.

File: Intro to open source with GitHub/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# How to work on opensource contribution with Git
+
+## Making a contribution
+1. Fork the repo which you want to work on (eg. https://github.com/take2rohit/sih_number_plate is original repo where you want to contribute) 
+1. Clone the forked repository (not the above repo link)
+`git clone https://github.com/your_userid/sih_number_plate`
+3. Do your changes whatever you wish and commit 
+`git add .`
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:8]: Some new people might not know. Maybe we can replace with '`git add file1 file2 file3` to add particular files and `git add .` to add all files'. Also, they have to do in the root folder of the repo (is it always clear?).

File: Intro to open source with GitHub/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+`git commit -m "Adding my contribution"`
+4. Push changes to GitHub. This updates your forked repo.
+`git push origin master`
+5. Create a pull request
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:12]: maybe some website link which shows the process. I know it's not difficult, but is it easy for new people?

File: Intro to open source with GitHub/README.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+6. If the owner find yours changes correct, he will merge it with his original repo.
+7. Congo! You just did your first contribution to original repo
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:14]: Congratulations! You just contributed to the original repo.

File: Intro to open source with GitHub/README.md [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+1. Fork the repo which you want to work on (eg. https://github.com/take2rohit/sih_number_plate is original repo where you want to contribute) 
+1. Clone the forked repository (not the above repo link)
+`git clone https://github.com/your_userid/sih_number_plate`
+3. Do your changes whatever you wish and commit 
+`git add .`
+`git commit -m "Adding my contribution"`
+4. Push changes to GitHub. This updates your forked repo.
+`git push origin master`
+5. Create a pull request
+6. If the owner find yours changes correct, he will merge it with his original repo.
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:13]: If the owner finds **your** changes correct, ......

File: Intro to open source with GitHub/README.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+7. Congo! You just did your first contribution to original repo
+
+## Creating a new contribution to sane repo
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:16]: Creating a new contribution to **the same** repo

File: Intro to open source with GitHub/README.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+1. Delete the old fork(if the master is ahed of your fok)
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:17]: Delete the old fork (if the master is **ahead** of your **fork**).

File: Intro to open source with GitHub/README.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+2. Create a new fork and do all changes there
+3. Thats it!
```
Comment (akshaykulkarni07) [inline Intro to open source with GitHub/README.md:19]: Create a new PR? 😂

**Review by akshaykulkarni07 — COMMENTED** [review]
@take2rohit 
Make the changes and let me know. Also, change the outer folder name to `opensource_git_contrib` or something compact.

File: Intro to open source with GitHub/README.md (hunk 2 above)
Comment (take2rohit) [reply Intro to open source with GitHub/README.md:4]: I dont think its a good idea.

File: Intro to open source with GitHub/README.md (hunk 5 above)
Comment (take2rohit) [reply Intro to open source with GitHub/README.md:8]: If someone is contributing its expected he should know git add

File: Intro to open source with GitHub/README.md (hunk 6 above)
Comment (take2rohit) [reply Intro to open source with GitHub/README.md:12]: will add

**Comment by take2rohit** [comment]
@akshaykvnit changes have been made. Check a780074 and merge

File: Intro to open source with GitHub/README.md (hunk 3 above)
Comment (take2rohit) [reply Intro to open source with GitHub/README.md:5]: numbering doesnt matters!

File: opensource_git_contrib/README.md [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+3. Do the changes and commit.
+`git add file1 file2`
+`git commit -m "Adding my contribution"`
+4. Push changes to GitHub. This updates your forked repo.
+`git push origin master`
+5. Create a pull request.
+6. If the owner finds your changes correct, he will merge it with his original repo.
+7. Congratulations! You just contributed to the original repo.
+
+Resources:
```
Comment (akshaykulkarni07) [inline opensource_git_contrib/README.md]: resources missing.

File: opensource_git_contrib/README.md [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+## Creating a new contribution to the sane repo
+1. Delete the old fork (if the master is ahead of your fork).
+2. Create a new fork and do all changes there.
+3. Create a Pull request and repeat above steps.
+4. Thats it!
+
+This is an oversimplified way of contributing
+For more advanced usage we can use branches and commands like adding upstream url and doing `git fetch`. You can google it learn for there itself.
```
Comment (akshaykulkarni07) [inline opensource_git_contrib/README.md]: something's wrong with the last sentence.

**Review by akshaykulkarni07 — COMMENTED** [review]
@take2rohit Just 2 comments. Rest seems fine.

**Comment by take2rohit** [comment]
> @take2rohit Just 2 comments. Rest seems fine.

Done!

**Comment by akshaykvnit** [comment]
@take2rohit the other folder is still there. Delete that with another pull request immediately.

**Pull request merged**